pkg/cacheutil/redis_client.go

```diff
@@ -219,9 +238,8 @@ func NewRedisClientWithConfig(logger log.Logger, name string, config RedisClient
 // SetAsync implement RemoteCacheClient.
 func (c *RedisClient) SetAsync(ctx context.Context, key string, value []byte, ttl time.Duration) error {
 	start := time.Now()
-	if _, err := c.Set(ctx, key, value, ttl).Result(); err != nil {
-		level.Warn(c.logger).Log("msg", "failed to set item into redis", "err", err, "key", key,
-			"value_size", len(value))
+	if err := c.client.Do(ctx, c.client.B().Set().Key(key).Value(rueidis.BinaryString(value)).ExSeconds(int64(ttl.Seconds())).Build()).Error(); err != nil {
```

Comment:
Why here we use `Set` with `ExSeconds` being added later instead of `Setex`, like in the `SetMulti` function? Would be cool to be consistent.

Reply:
Also, wouldn't it be wise to use `DoCache` here to populate the client side cache too? I don't see `DoCache` or `DoMultiCache` being used anymore and from what I understand from Rueidis' README those are functions that will also populate client side cache.

Reply:
I noticed there is `MGetCache` for getting through client side cache, but I fail to see where we write to the client side cache.

Reply:
Hi @douglascamata,

The population of client side cache is done by rueidis when you get response from `DoCache` and `DoMultiCache` with supported commands, such as GET and MGET.

SET is not supported by `DoCache` by design, because the client must be tracked by redis server via cacheable requests to receive related invalidation notifications.

Reply:
@rueian gotcha, thanks for the clarification! 

I thought it would be useful for us to populate the client side cache whenever we SET something, but it if cost is another GET and by consequence of the design, might not be a good idea.